src/integration-tests/src/main/java/com/microsoft/azure/eventhubs/checkstatus/ReceiveByDateTime.java

```diff
@@ -67,82 +71,69 @@ public static void main(String[] args)
-
-                                        if (eventDateTime.isAfter(startTime)) {
-                                            System.out.println("Found a processed alert: " + dataString);
-                                            System.exit(0);
-                                        }
-                                    } catch (Exception e) {
-                                        System.out.println("There was a problem parsing the date of the event: "
-                                                + receivedEvent.getBytes());
-                                        continue;
-                                    }
-                                }
-                                batchSize++;
-                            }
-                        }
-
-                        System.out.println(String.format("ReceivedBatch Size: %s", batchSize));
-                    }, executorService).get();
+        final String connStr = String.format("Endpoint=sb://%s.servicebus.windows.net/;", namespaceName) +
+                               String.format("EntityPath=%s;", eventHubName) +
+                               String.format("SharedAccessKeyName=RootManageSharedAccessKey;", sasKeyName) +
+                               String.format("SharedAccessKey=${eventhub_key}", sasKey);
+       
+        EventHubConsumerAsyncClient consumer = new EventHubClientBuilder()
+            .connectionString(connStr)
+            .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
+            .buildAsyncConsumerClient();
+
+        Disposable subscription  = consumer.getPartitionIds().take(1)
+        .flatMap(partitionId -> consumer.receiveFromPartition(partitionId, EventPosition.fromEnqueuedTime(Instant.EPOCH)))
+        .subscribe(partitionEvent -> {
+            EventData event = partitionEvent.getData();
+            PartitionContext partitionContext = partitionEvent.getPartitionContext();
+
+            System.out.print(String.format("[%s] Offset: %s, #: %s, Time: %s, PT: %s, ",
+            new java.util.Date(), event.getOffset(),
+            event.getSequenceNumber(),
+            event.getEnqueuedTime(), partitionContext.getPartitionId()));
+            if (event.getBody() != null) {
+
+                try {
+                    final String dataString = new String(event.getBody(), UTF_8);
+                    final JSONObject obj = new JSONObject(dataString);
+                    final String windowStartString = obj.getString("windowStart");
+                    final LocalDateTime eventDateTime = LocalDateTime.parse(windowStartString,
+                            formatter);
+                    System.out.println("Event Time: " + windowStartString);
+
+                    if (eventDateTime.isAfter(startTime)) {
+                        System.out.println("Found a processed alert: " + dataString);
+                        System.exit(0);
+                    }
+                    countDownLatch.countDown();
```

Comment:
Countdown latch is probably not the item you want.. from the original sample, it looks like we are just consuming events until 15 minutes have elapsed.

Reply:
A more reactive way of doing this would probably be:
```java
Duration applicationDuration = Duration.ofMinutes(15);

// We get all the partitions and take the first partition to create a consumer from. The consumer 
// begins listening at the start of the stream.
// This is asynchronous. The program will move to the next line after setting up the subscription.
Disposable subscription = consumer.getPartitionIds()
    .take(1)
    .flatMap(partitionId -> consumer.receiveFromPartition(partitionId, EventPosition.earliest()))
    .take(applicationDuration)
    .subscribe(partitionEvent -> {
            // ... 
        }, error -> {
            // onError
        }, () -> {
            // onComplete
        });

try {
    // Sleep the application so it does not end. This is because the .subscribe() is a non-blocking
    // call. While the thread is sleeping, the subscription will run in the background, processing 
    // messages.
    Thread.sleep(applicationDuration.toMillis());
} catch (InterruptedException e) {
    System.err.println("Unable to sleep: " + e);
} finally {
    subscription.dispose();
    consumer.dispose();
}

System.out.println("Complete");
```